master/master.go

```diff
@@ -133,6 +198,114 @@ func SendTask(t *lib.Task, ip string) (*http.Response, error) {
 	return http.Post(url, lib.BodyType, bytes.NewBuffer(body))
 }
 
+func newEC2() *ec2.EC2 {
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(lib.AWSRegion)},
+	)
+	if err != nil {
+		panic(err)
+	}
+	return ec2.New(sess)
+}
+
+func (m *Master) initAWS() {
+	m.svc = newEC2()
+
+	// create one slave on startup
+	s, err := createSlaves(m.svc, 1)
+	if err != nil {
+		fmt.Println("Failed to create slave")
+		panic(err)
+	}
+
+	ip := getPublicIP(m.svc, s[0])
+	// probably not necessary becase
+	if ip != nil {
+		m.instances[*ip] = slave{
+			maxSlots: lib.MaxSlotsPerInstance,
+			instance: s[0],
+		}
+	}
+}
+
+func getPublicIP(svc *ec2.EC2, instance *ec2.Instance) *string {
+	params := ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name: aws.String("instance-state-name"),
+				Values: []*string{
+					aws.String("pending"),
+					aws.String("running"),
+				},
+			},
+		},
+		InstanceIds: []*string{
+			instance.InstanceId,
+		},
+	}
+
+	var i int
+	for {
+		res, err := svc.DescribeInstances(&params)
+		fmt.Println("reservations", len(res.Reservations))
+
+		// ignore the error because we may try again
+		if err == nil &&
+			len(res.Reservations) == 1 &&
+			len(res.Reservations[0].Instances) == 1 {
+
+			fmt.Println(res.Reservations[0].Instances[0].State)
+			if res.Reservations[0].Instances[0].PublicIpAddress != nil {
+				fmt.Println("Found public IP")
+				return res.Reservations[0].Instances[0].PublicIpAddress
+			}
+
+		}
+		time.Sleep(10 * time.Second)
```

Comment:
No, this function is just there for testing, we won't need it when heartbeats are available.